src/uk/gov/hmcts/contino/GithubAPI.groovy

```diff
@@ -13,22 +13,94 @@ class GithubAPI {
     this.steps = steps
   }
 
+  private static cachedLabelList = [
+    'isValid': false,
+    'cache': []
+  ]
+
+  def currentProject() {
+    return new RepositoryUrl().getShort(this.steps.env.CHANGE_URL)
+  }
+
+  def currentPullRequestNumber() {
+    return this.steps.env.CHANGE_ID
+  }
+
   /**
-   * Add labels to the current pull request.  MUST be run with an onPR() closure.
-   *
-   * @param labels
-   *   A List of labels
+   * Check for Pull Request
    */
-  def addLabelsToCurrentPR(labels) {
+  private isPR(String branch_name) {
+    def isPr = new ProjectBranch(branch_name).isPR()
+    this.steps.echo "Is ${branch_name} a PR?: ${isPr}"
+    return isPr
+  }
+
+  /**
+   * Return whether the cache is valid
+   */
+  def static isCacheValid() {
+    return cachedLabelList.isValid
+  }
+
+  /**
+   * Return whether the cache is empty
+   */
+  def static isCacheEmpty() {
+    return cachedLabelList.cache.isEmpty()
+  }
+
+  /**
+   * Return cache contents as list
+   */
+  def static getCache() {
+    return cachedLabelList.cache
+  }
+
+  /**
+   * Clears this.cachedLabelList
+   */
+  void clearLabelCache() {
```

Comment:
is this still needed? is it a future like method? I can't see it called outside of tests

Reply:
It's essential to clear up the cache between tests.  May as well leave it in place for future use if required - we have an exposed method for refreshing the cache, and I think having one to clear it makes sense.